### PR TITLE
[IE CLDNN] Gather 5d/6d support

### PR DESCRIPTION
**Pull request by ldebski**
Addition of 5d and 6d support for gather primitive.
I also deleted changes introduced by this WA since it's no longer needed: https://github.com/openvinotoolkit/openvino/pull/1056 

JIRA: CVS-33896

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/gather/gather_kernel_ref.cpp [hunk 1]
```diff
@@ -145,7 +188,16 @@ JitConstants GatherKernelRef::GetJitConstants(const gather_params& params) const
...
     if (!params.fused_ops.empty()) {
-        FusedOpsConfiguration conf = { "", {"b", "f", "y", "x"}, "val", params.inputs[0].GetDType() };
+        std::vector<std::string> idx_order;
+        if (params.inputs[0].GetDims().size() <= 4) {
+            idx_order = {"b", "f", "y", "x"};
+        } else if (params.inputs[0].GetDims().size() == 5) {
+            idx_order = {"b", "f", "z", "y", "x"};
+        } else if (params.inputs[0].GetDims().size() == 6) {
+            idx_order = {"b", "f", "w", "z", "y", "x"};
+        }
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/gather/gather_kernel_ref.cpp:198]: Can we use `std::vector<std::string> GetOrder(size_t size)` function here?

File: inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp [hunk 2]
```diff
@@ -4989,12 +5005,24 @@ INSTANTIATE_TEST_CASE_P(fusings_gpu, gather_quantize,
                         gather_test_params{ CASE_GATHER_FP32_3, 2, 3 },
                         gather_test_params{ CASE_GATHER_FP32_4, 2, 3 },
                         gather_test_params{ CASE_GATHER_FP32_5, 2, 3 },
-
+
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp:5008]: Please remove extra spaces

File: inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp [hunk 3]
```diff
@@ -5020,12 +5048,24 @@ INSTANTIATE_TEST_CASE_P(fusings_gpu, gather_scale_activation,
                         gather_test_params{ CASE_GATHER_FP32_3, 2, 4 },
                         gather_test_params{ CASE_GATHER_FP32_4, 2, 4 },
                         gather_test_params{ CASE_GATHER_FP32_5, 2, 4 },
-
+
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp:5051]: Same here

File: inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/gather/gather_kernel_ref.cpp (hunk 1 above)
Comment (ldebski) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/gather/gather_kernel_ref.cpp]: Done

File: inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp (hunk 2 above)
Comment (ldebski) [reply inference-engine/thirdparty/clDNN/tests/test_cases/fusings_gpu_test.cpp]: Done

**Comment by sshlyapn** [comment]
You have some failed Gather test, please fix it before merge

**Pull request merged**